openedx/core/djangoapps/credit/tests/test_api.py

```diff
@@ -876,3 +1070,126 @@ def _assert_credit_status(self, expected_status):
         statuses = api.get_credit_requests_for_user(self.USER_INFO["username"])
         self.assertEqual(statuses[0]["status"], expected_status)
+
+
+@unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in LMS')
+@override_settings(ECOMMERCE_API_URL=TEST_API_URL, ECOMMERCE_API_SIGNING_KEY=TEST_API_SIGNING_KEY)
+@ddt.ddt
+class CourseApiTests(CreditApiTestBase):
+    """Test Python API for course product information."""
+    def setUp(self):
+        super(CourseApiTests, self).setUp()
+        self.user = UserFactory()
+        self.add_credit_course(self.course_key)
+        self.credit_config = CreditApiConfig(cache_ttl=100, enabled=True)
+        self.credit_config.save()
+
+        # Add another provider.
+        CreditProvider.objects.get_or_create(
+            provider_id='ASU',
+            display_name='Arizona State University',
+            provider_url=self.PROVIDER_URL,
+            provider_status_url=self.PROVIDER_STATUS_URL,
+            provider_description=self.PROVIDER_DESCRIPTION,
+            enable_integration=self.ENABLE_INTEGRATION,
+            fulfillment_instructions=self.FULFILLMENT_INSTRUCTIONS,
+            thumbnail_url=self.THUMBNAIL_URL
+        )
+
+    @httpretty.activate
+    def test_get_credit_providers_method(self):
+        """Verify that parsed providers list is returns after getting course production information."""
+        user = UserFactory.create()
+        self._mock_ecommerce_courses_api(self.course_key, self.COURSE_API_RESPONSE, 200)
+        response_providers = get_credit_providers_by_course(user, self.course_key)
+        self.assertEqual(self.PROVIDERS_LIST, response_providers)
+
+    @httpretty.activate
+    @mock.patch('edx_rest_api_client.client.EdxRestApiClient.__init__')
+    def test_get_credit_providers_method_with_exception(self, mock_init):
+        """Verify that in case of any exception it logs the error and return."""
+        mock_init.side_effect = Exception
+        response = get_credit_providers_by_course(self.user, self.course_key)
+        self.assertTrue(mock_init.called)
+        self.assertEqual(response, None)
+
+    @httpretty.activate
+    def test_get_credit_providers_caching(self):
+        """Verify that providers list is cached."""
+        self._mock_ecommerce_courses_api(self.course_key, self.COURSE_API_RESPONSE, 200)
+        user = UserFactory.create()
+        # Warm up the cache.
+        response_providers = get_credit_providers_by_course(user, self.course_key)
+        self.assertEqual(self.PROVIDERS_LIST, response_providers)
+
+        # Hit the cache.
+        response_providers = get_credit_providers_by_course(user, self.course_key)
+        self.assertEqual(self.PROVIDERS_LIST, response_providers)
+
+        # Verify only one request was made.
+        self.assertEqual(len(httpretty.httpretty.latest_requests), 1)
+
+        # # Hit the api with clearing cache.
+        for __ in range(3):
```

Comment:
I don't think this is necessary. The goal of this test is to verify that caching works, which it already has by this point.